Framework/Geometry/src/Instrument/GridDetector.cpp

```diff
@@ -0,0 +1,939 @@
+    return;
+  if (yIndex >= ypixels())
+    return;
+
+  // TODO: Do I need to put something smart here for the first 3 parameters?
+  auto comp = getAtXYZ(xIndex, yIndex, 0);
+  testRay.addLink(intersec, intersec, 0.0, *(comp->shape()),
+                  comp->getComponentID());
+}
+
+//-------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------
+// ------------ IObjComponent methods ----------------
+//-------------------------------------------------------------------------------------------------
+//-------------------------------------------------------------------------------------------------
+
+//-------------------------------------------------------------------------------------------------
+/// Does the point given lie within this object component?
+bool GridDetector::isValid(const V3D &) const {
+  throw Kernel::Exception::NotImplementedError(
+      "GridDetector::isValid() is not implemented.");
+}
+
+//-------------------------------------------------------------------------------------------------
+/// Does the point given lie on the surface of this object component?
+bool GridDetector::isOnSide(const V3D &) const {
+  throw Kernel::Exception::NotImplementedError(
+      "GridDetector::isOnSide() is not implemented.");
+}
+
+//-------------------------------------------------------------------------------------------------
+/// Checks whether the track given will pass through this Component.
+int GridDetector::interceptSurface(Track &) const {
+  throw Kernel::Exception::NotImplementedError(
+      "GridDetector::interceptSurface() is not implemented.");
+}
+
+//-------------------------------------------------------------------------------------------------
+/// Finds the approximate solid angle covered by the component when viewed from
+/// the point given
+double GridDetector::solidAngle(const V3D &) const {
+  throw Kernel::Exception::NotImplementedError(
+      "GridDetector::solidAngle() is not implemented.");
+}
+
+//-------------------------------------------------------------------------------------------------
+/// Try to find a point that lies within (or on) the object
+int GridDetector::getPointInObject(V3D &) const {
+  throw Kernel::Exception::NotImplementedError(
+      "GridDetector::getPointInObject() is not implemented.");
+}
+
+//-------------------------------------------------------------------------------------------------
+/**
+ * Get the bounding box and store it in the given object. This is cached after
+ * the first call.
+ * @param assemblyBox :: A BoundingBox object that will be overwritten
+ */
+void GridDetector::getBoundingBox(BoundingBox &assemblyBox) const {
+  if (m_map) {
+    if (hasComponentInfo()) {
+      assemblyBox = m_map->componentInfo().boundingBox(index(), &assemblyBox);
+      return;
+    }
+  }
+  if (!m_cachedBoundingBox) {
```

Comment:
Is the cache really necessary?

Reply:
Not really. Do you have a strong desire for me to remove it?

Reply:
When this gets to production, unless the optimisation is needed, i would not keep it there.